### DISCUSSION TITLE
Multiple Obligations

### DISCUSSION THREAD

**Discussion by zeno521** [authzforce/core#90] [Q&A]
Hi AuthzForce team,

We are working on a POC with Authzforce PDP engine. We have multiple permit rules (each with obligations) in a policy with **deny overrides** rule combining algorithm. We expected the obligations from all rules to be returned in the results when all the rules and policy return Permit. However, we can only see the obligations returned from the first permit rule.

Could you indicate if this is an expected behaviour please? If it is, how to return the obligations from all the rules?

Your earliest response is highly appreciated!

**Comment by cdanger** [comment]
Which version of AuthzForce Core are you using?

I tried with these Policy and Request:
 [Policy.xml.txt](https://github.com/authzforce/core/files/13326708/Policy.xml.txt) 
[Request.xml.txt](https://github.com/authzforce/core/files/13326707/Request.xml.txt)

Works for me. I get all obligations from the two Permit Rules in the Result.

**Comment by zeno521** [comment]
We are using v24.1.0.

We had a try with your sample policy and request, and it returned the obligations from the two Permit rules as expected. However, if we remove the Deny rule (rule 1), it only returned obligations from the first Permit rule. Is that correct behaviour please?

**Comment by cdanger** [comment]
There is no v24.1.0 for AuthzForce core. Latest [release](https://github.com/authzforce/core/releases) is **21.0.0**. Are you sure?

**Comment by zeno521** [comment]
Sorry, it's 21.4.0.

https://mvnrepository.com/artifact/org.ow2.authzforce/authzforce-ce-core-pdp-api/21.4.0

**Comment by cdanger** [comment]
OK this is the version of authzforce-ce-core-pdp-api (Core PDP API). But which version of authzforce-ce-core-pdp-cli (if you use the [CLI](https://github.com/authzforce/core#cli) ) or authzforce-ce-core-pdp-engine (if you use the [Java API](https://github.com/authzforce/core#java-api)  ) do you use? 

See the [Usage section of the README](https://github.com/authzforce/core#usage) for more info.

**Comment by zeno521** [comment]
authzforce-ce-core-pdp-engine-20.3.2.jar

**Comment by cdanger** [comment]
Bug confirmed. I create the issue #92 .

**Comment by zeno521** [comment]
Thanks @cdanger! May I know when the fix is expected to be available please?

**Comment by cdanger** [comment]
Should be fixed now in latest release 21.0.1.

**Comment by zeno521** [comment]
Awesome! Thank you very much!
